### PR TITLE
Exam

### PR DESCRIPTION
**Pull request by jpcosal**

### CODE REVIEW

File: Exam/src/exercice2/LogItem.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	public static String format() {
+		String msg = "";
+		msg += timestamp.getDay()+"/";
+		msg += timestamp.getMonth()+"/";
+		msg += timestamp.getYear()+"-";
+		msg += timestamp.getHours()+":";
+		msg += timestamp.getMinutes()+"-";
+		msg += level+"-";
+		msg += message+".";
+		return msg;
```
Comment (paulmichels) [inline Exam/src/exercice2/LogItem.java]: Tu peux utiliser un SimpleDateFormat pour pas te prendre la tête

File: Exam/src/exercice4/CaisseTest.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		
+		Article asterix = new Article("Asterix", 7.99f);
+		livre.addDisponibilite(asterix);
+		Article boule = new Article("Boule&Bill", 5.99f);
+		livre.addDisponibilite(boule);
+		
+		Article monde = new Article("Le Monde", 3.99f);
+		presse.addDisponibilite(monde);
+		
+		Article parisien = new Article("Le Parisien", 20.99f);
```
Comment (paulmichels) [inline Exam/src/exercice4/CaisseTest.java]: oof

File: Exam/src/exercice1/Somme.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+			result += i;
+		}
+		result = result*result;
+		return result;
+	}
+	
+	public static int Difference(int nombre) {
+		int result = 0;
+		int SommeCarres = SommeDesCarres(nombre);
+		int CarreSomme = CarreDeLaSomme(10);
```
Comment (paulmichels) [inline Exam/src/exercice1/Somme.java]: Erreur ici

File: Exam/src/exercice3/Jouet.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+			desc += listFormes.get(i).type+";";
+		}
+		return desc;
+	}
+	
+	public float CalculAire() {
+		float aire = 0;
+		for (int i=0; i<listFormes.size(); i++){
+		   if (listFormes.get(i).type == "cercle") {
+			   aire += Math.PI * (listFormes.get(i).rayon * listFormes.get(i).rayon);
```
Comment (jpcosal) [inline Exam/src/exercice3/Jouet.java]: This should belong to a child class...

File: Exam/src/exercice4/Article.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package exercice4;
+
+public class Article {
+	String nomArticle;
```
Comment (jpcosal) [inline Exam/src/exercice4/Article.java]: no need for that

File: Exam/src/exercice4/Compte.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
+package exercice4;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Compte {
+	String nomCompte;
+	protected List<Article> listArticles = new ArrayList<Article>();
```
Comment (jpcosal) [inline Exam/src/exercice4/Compte.java]: Hmmm... No... Your design makes `Article` a component of `Vente`, so you just need `listVentes`

File: Exam/src/exercice4/Compte.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+	
+	Compte (String _nomCompte) {
+		this.nomCompte = _nomCompte;
+	}
+	
+	public void addVente(Vente vente) {
+		listVentes.add(vente);
+	}
+	
+	public void addDisponibilite(Article article) {
```
Comment (jpcosal) [inline Exam/src/exercice4/Compte.java]: What would that be for??

**Pull request closed without merging**